File: dataset/output/MISTRAL_codestral-latest/Task6_MISTRAL_codestral-latest.cpp
#include <vector>
#include <unordered_map>

double findUniq(std::vector<double> arr) {
    std::unordered_map<double, int> count;
    for (double num : arr) {
        count[num]++;
    }
    for (auto const& entry : count) {
        if (entry.second == 1) {
            return entry.first;
        }
    }
    return 0;
}

File: dataset/output/MISTRAL_codestral-latest/task6/Task6_MISTRAL_codestral_latest.java
public class Task6_MISTRAL_codestral_latest {
    public static double findUniq(double arr[]) {
        double a = arr[0];
        double b = arr[1];
        for (int i = 2; i < arr.length; i++) {
            if (arr[i] != a) {
                b = arr[i];
                break;
            }
        }
        return a if count(arr, a) == 1 else b;
    }

    public static int count(double arr[], double x) {
        int count = 0;
        for (int i = 0; i < arr.length; i++) {
            if (arr[i] == x) {
                count++;
            }
        }
        return count;
    }
}